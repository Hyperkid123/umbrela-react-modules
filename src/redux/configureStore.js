import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
