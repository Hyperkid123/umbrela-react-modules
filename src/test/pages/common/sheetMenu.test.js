import React from 'react'
import SheetMenu, {SheetMenu as Snapshot} from '../../../pages/common/sheetMenu';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('Sheet menu component', () => {
  const initialProps = {};
  beforeEach(() => {
    initialProps.sheets = [];
    initialProps.researchId = 0;
    initialProps.activeSheetId = 0;
    initialProps.draggingElement = false;
    initialProps.translate = jest.fn();
    initialProps.createNewSheet = jest.fn();
    initialProps.selectEditorSheet = jest.fn();
    initialProps.dragSheetCard = jest.fn();
    initialProps.remapSheets = jest.fn();
    initialProps.preventDrag = false;
  });

  it('should render correclty empty', () => {
    const tree = shallow(
      <Snapshot {...initialProps}/>
    )
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should render loading heading', () => {
    const tree = shallow(
      <Snapshot {...initialProps} sheets={false}/>
    )
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should call dragSheetCard action on move card', () => {
    const dragSheetCard = jest.fn();
    const wrapper = shallow(
      <Snapshot {...initialProps} dragSheetCard={dragSheetCard}/>
    )
    wrapper.instance().moveCard(10, 99);
    expect(dragSheetCard).toHaveBeenCalledWith(10, 99);
  });

  it('should render correclty with draggable menu', () => {
    const sheets = [{
      sheetId: 0,
      title: 'sheet 0'
    }, {
      sheetId: 1,
      title: 'sheet 1'
    }]
    const tree = shallow(
      <Snapshot {...initialProps} sheets={sheets}/>
    )
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should render correclty with non draggable menu', () => {
    const sheets = [{
      sheetId: 0,
      title: 'sheet 0'
    }, {
      sheetId: 1,
      title: 'sheet 1'
    }]
    const tree = shallow(
      <Snapshot {...initialProps} sheets={sheets} preventDrag={true}/>
    )
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should render correclty empty withnout new sheet button', () => {
    const tree = shallow(
      <Snapshot {...initialProps} hideNewSheet={true}/>
    )
    expect(toJson(tree)).toMatchSnapshot();
  });
});
