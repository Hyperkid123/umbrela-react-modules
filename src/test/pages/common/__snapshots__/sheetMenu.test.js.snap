// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sheet menu component should render correclty empty 1`] = `
<styled.div>
  <styled.div />
  <styled.div>
    <WithStyles(Button)
      color="primary"
      onClick={[Function]}
    >
      <pure(Add) />
    </WithStyles(Button)>
    <styled.ul />
  </styled.div>
</styled.div>
`;

exports[`Sheet menu component should render correclty empty withnout new sheet button 1`] = `
<styled.div>
  <styled.div />
  <styled.div>
    <styled.ul />
  </styled.div>
</styled.div>
`;

exports[`Sheet menu component should render correclty with draggable menu 1`] = `
<styled.div>
  <styled.div />
  <styled.div>
    <WithStyles(Button)
      color="primary"
      onClick={[Function]}
    >
      <pure(Add) />
    </WithStyles(Button)>
    <styled.ul>
      <DragSource(DropTarget(SheetDraggableCard))
        index={0}
        key="0"
        moveCard={[Function]}
        onDragEnd={[Function]}
        preventDrag={false}
        researchId={0}
      >
        <MenuListItem
          active={true}
          dragging={false}
          key="0"
          label="sheet 0"
          onClick={[Function]}
          preventDrag={false}
          sheetId={0}
        />
      </DragSource(DropTarget(SheetDraggableCard))>
      <DragSource(DropTarget(SheetDraggableCard))
        index={1}
        key="1"
        moveCard={[Function]}
        onDragEnd={[Function]}
        preventDrag={false}
        researchId={0}
      >
        <MenuListItem
          active={false}
          dragging={false}
          key="1"
          label="sheet 1"
          onClick={[Function]}
          preventDrag={false}
          sheetId={1}
        />
      </DragSource(DropTarget(SheetDraggableCard))>
    </styled.ul>
  </styled.div>
</styled.div>
`;

exports[`Sheet menu component should render correclty with non draggable menu 1`] = `
<styled.div>
  <styled.div />
  <styled.div>
    <WithStyles(Button)
      color="primary"
      onClick={[Function]}
    >
      <pure(Add) />
    </WithStyles(Button)>
    <styled.ul>
      <MenuListItem
        active={true}
        dragging={false}
        key="0"
        label="sheet 0"
        onClick={[Function]}
        preventDrag={true}
        sheetId={0}
      />
      <MenuListItem
        active={false}
        dragging={false}
        key="1"
        label="sheet 1"
        onClick={[Function]}
        preventDrag={true}
        sheetId={1}
      />
    </styled.ul>
  </styled.div>
</styled.div>
`;

exports[`Sheet menu component should render loading heading 1`] = `
<styled.h2>
  Loading
</styled.h2>
`;
